snowpack/src/commands/dev.ts

```diff
@@ -120,6 +121,44 @@ class NotFoundError extends Error {
   }
 }
 
+const noOpHeadersTransformer: HeadersTransformer = (_req, proposed) => proposed;
+
+/**
+ * Decorates res.writeHead, such that a user-supplied HeadersTransformer may inspect
+ * the proposed response headers, and (with reference to the request): optionally propose
+ * alternative headers.
+ * This enables users to apply their own CORS or security policies, and to override decisions
+ * about Content-Type.
+ */
+function decorateWriteHead(
+  req: http.IncomingMessage,
+  res: http.ServerResponse,
+  headersTransformer: HeadersTransformer
+): void {
+  const original = res.writeHead;
+  const decorated: http.ServerResponse['writeHead'] =
+  // writeHead has two overloads, so our method signature is the (messy) union of those
+  function writeHead(
```

Comment:
For stacktrace friendliness, I'd recommend renaming this function to something like `customizedWriteHead`, which is how it can appear in call stacks without getting confused for the original `writeHead` method

Reply:
agreed. though I think I'll rewrite to avoid monkey-patching anyway, so this function will go away.